fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -350,6 +350,43 @@ ACTOR Future<ISimulator::KillType> simulatedFDBDRebooter(Reference<ClusterConnec
 	}
 }
 
+ACTOR Future<ISimulator::KillType> simulatedInvalidLocalityFDBDRebooter(
+    Reference<ClusterConnectionFile> connFile, IPAddress ip, bool sslEnabled, Reference<TLSOptions> tlsOptions,
+    uint16_t port, uint16_t listenPerProcess, LocalityData localities, ProcessClass processClass,
+    std::string* dataFolder, std::string* coordFolder, std::string baseFolder, ClusterConnectionString connStr,
+    bool useSeedFile, AgentMode runBackupAgents, std::string whitelistBinPaths) {
+	state int numReboots = deterministicRandom()->random01() * 5;
+	state int curReboot = 0;
+	// state ISimulator::KillType ret = ISimulator::KillType::None;
+	loop {
+		if (curReboot == numReboots) {
+			ISimulator::KillType ret = wait(simulatedFDBDRebooter(
+			    connFile, ip, sslEnabled, tlsOptions, port, listenPerProcess, localities, processClass, dataFolder,
+			    coordFolder, baseFolder, connStr, useSeedFile, runBackupAgents, whitelistBinPaths));
+			return ret;
+		} else if (curReboot < numReboots) {
+			// Simulate the case when operators try to misconfig localities multiple times
+			// ProcessId is unset on purpose because it will be set based on data filename
+			state LocalityData misConfLocalities(Optional<Standalone<StringRef>>(), Optional<Standalone<StringRef>>(),
+			                                     Optional<Standalone<StringRef>>(), Optional<Standalone<StringRef>>());
+			// Must set keyProcessId and machineId
+			misConfLocalities.set(LocalityData::keyProcessId, localities.get(LocalityData::keyProcessId));
+			misConfLocalities.set(LocalityData::keyMachineId, localities.get(LocalityData::keyMachineId));
+			for (int i = 0; i < localities.size(); i++) {
+				if (deterministicRandom()->random01() < 0.2) {
```

Comment:
It's still possible, with a very low possibility though, to have every entry inserted into `misConfLocalities`. Maybe compare the size of `misConfLocalities` and `localities` and randomly erase some entries in `misConfLocalities` if necessary.

Reply:
it's on purpose. It covers the situation when locality is set correctly in the first place. :)